api/src/main/java/org/codice/ddf/admin/api/validation/SourceValidationUtils.java

```diff
@@ -17,21 +17,55 @@
 import static org.codice.ddf.admin.api.config.sources.SourceConfiguration.SOURCE_USER_PASSWORD;
 import static org.codice.ddf.admin.api.handler.ConfigurationMessage.createInvalidFieldMsg;
 import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_FACTORY_PIDS;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_GMD_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_PROFILE_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_SPEC_FACTORY_PID;
 import static org.codice.ddf.admin.api.services.OpenSearchServiceProperties.OPENSEARCH_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS1_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS2_FACTORY_PID;
 import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS_FACTORY_PIDS;
 import static org.codice.ddf.admin.api.validation.ValidationUtils.validateString;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
+import org.apache.commons.lang.StringUtils;
 import org.codice.ddf.admin.api.config.sources.SourceConfiguration;
+import org.codice.ddf.admin.api.configurator.Configurator;
 import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
 
 public class SourceValidationUtils {
 
-    public static List<ConfigurationMessage> validateWfsFactoryPid(String factoryPid,
-            String configId) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceValidationUtils.class);
+
+    private static final List<String> FACTORY_PIDS = ImmutableList.of(CSW_GMD_FACTORY_PID,
+            CSW_PROFILE_FACTORY_PID,
+            CSW_SPEC_FACTORY_PID,
+            OPENSEARCH_FACTORY_PID,
+            WFS1_FACTORY_PID,
+            WFS2_FACTORY_PID);
+
+    private Configurator configurator;
+
+    public SourceValidationUtils() {
+        configurator = new Configurator();
+    }
+
+    /*
+        For testing purposes only.
+     */
+    protected SourceValidationUtils(Configurator configurator) {
+        this.configurator = configurator;
+    }
+
+    public List<ConfigurationMessage> validateWfsFactoryPid(String factoryPid, String configId) {
```

Comment:
My earlier comment then makes these methods -- where this utils class knows about specific source types and their factory pids -- a bad fit for a common class. You can extract out the common functionality and pass in an extra param for the list of factory pids that should be validated.